packages/core/test/request-builder/header-builder/header-builder.spec.ts

```diff
@@ -234,6 +236,47 @@ describe('Header-builder:', () => {
     expect(headers['x-csrf-token']).toBe('mocked-x-csrf-token');
   });
 
+  it('Skips csrf token retrieval for existing csrf header', async () => {
```

Comment:
General: What is the strategy to keep track on which tests relate to the new or old depricated methods. In case we clean up the deprecated ones it could become difficult to understand if thre is already a test covering it or you need to migrate. If see the `Prioritizes custom Authorization headers (upper case A)` kind of doulbled but some other test are only in the old spec.

Reply:
Let's discuss this in person. I agree that this is not optimal.